apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ir/OperationBasedModelGroupBuilder.kt

```diff
@@ -112,23 +114,44 @@ internal class OperationBasedModelGroupBuilder(
   /**
    * A grouping key for fragments
    */
-  private data class InlineFragmentKey(val typeCondition: String, val condition: BooleanExpression<BVariable>)
+  private data class InlineFragmentKey(val typeCondition: String, val condition: BooleanExpression<BTerm>)
 
-  private fun BooleanExpression<BVariable>.toName(): String = when (this) {
+  private fun BooleanExpression<BTerm>.toName(): String = when (this) {
     is BooleanExpression.True -> "True"
     is BooleanExpression.False -> "False"
     is BooleanExpression.And -> this.operands.joinToString("And") { it.toName() }
     is BooleanExpression.Or -> this.operands.joinToString("Or") { it.toName() }
     is BooleanExpression.Not -> "Not${this.operand.toName()}"
-    is BooleanExpression.Element -> this.value.name.capitalizeFirstLetter()
-    else -> error("")
+    is BooleanExpression.Element -> {
+      val bVariable = this.value as? BVariable ?: error("Unexpected term type in toName")
+      bVariable.name.capitalizeFirstLetter()
+    }
+    else -> error("Unexpected boolean expression type in toName")
   }
 
   private fun InlineFragmentKey.toName(): String = buildString {
     append(typeCondition.capitalizeFirstLetter())
     if (condition != BooleanExpression.True) {
-      append("If")
-      append(condition.toName())
+      // If at least one condition is a BLabel, it is enough to make a unique name, because the label is unique
+      val deferCondition: BLabel? = condition.firstElementOfType(BLabel::class)
+      if (deferCondition != null) {
+        append("Defer")
```

Comment:
I would just skip "Defer" there. The [samples from the wg](https://github.com/graphql/graphql-wg/blob/main/rfcs/DeferStream.md) already include "defer" so that'd be a lot of "defer"

Reply:
This would be ambiguous if there's no Label though. 

```graphql
computers {
   ... on Computer { id }  # 'onComputer'
   ... on Computer @defer { cpu }  # also 'onComputer'
}
```

But we could append `Defer` only in that case?

Reply:
> This would be ambiguous if there's no Label though.

Right!

> But we could append Defer only in that case?

Except if someone does `@defer(label: "defer")` 🤪 

Oh man... Ok let's stick with `"Defer"`